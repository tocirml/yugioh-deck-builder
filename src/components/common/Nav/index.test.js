import React from 'react';
import { Nav } from './index';
import { NavLink, MemoryRouter } from 'react-router-dom';

import { render, screen } from '@testing-library/react';
import '@testing-library/jest-dom/extend-expect';

import { shallow } from 'enzyme';

const links = {
  HomeLinkText: 'Home',
  CardListLinkText: 'Card List',
  DeckBuilderLinkText: 'Deck Builder',
  AboutLinkText: 'About',
};

describe('rendering', () => {
  it('renders without crashing', () => {
    shallow(<Nav />);
  });

  it('renders 4 NavLink', () => {
    const wrapper = shallow(<Nav />);
    expect(wrapper.find(NavLink).length).toEqual(4);
  });

  it('renders links text correctly', () => {
    render(
      <MemoryRouter>
        <Nav />
      </MemoryRouter>
    );

    // screen.debug(screen.getByText(links.HomeLinkText));

    screen.getByText(links.HomeLinkText);
    screen.getByText(links.CardListLinkText);
    screen.getByText(links.DeckBuilderLinkText);
    screen.getByText(links.AboutLinkText);
  });
});

describe('mobile', () => {
  it('has mobile class if passed', () => {
    render(
      <MemoryRouter>
        <Nav mobile={true} />
      </MemoryRouter>
    );

    expect(screen.getByTestId('navbar')).toHaveClass('mobile');
  });
});

describe('correct links', () => {
  const { getByText } = render(
    <MemoryRouter>
      <Nav />
    </MemoryRouter>
  );

  const homeLink = getByText(links.HomeLinkText);
  const cardListLink = getByText(links.CardListLinkText);
  const deckBuilderLink = getByText(links.DeckBuilderLinkText);
  const aboutLink = getByText(links.AboutLinkText);

  it('Home has correct href', () => {
    expect(homeLink).toHaveAttribute('href', '/');
  });

  it('Card List has correct href', () => {
    expect(cardListLink).toHaveAttribute('href', '/card-list');
  });

  it('Deck Builder has correct href', () => {
    expect(deckBuilderLink).toHaveAttribute('href', '/deck-builder');
  });

  it('About has correct href', () => {
    expect(aboutLink).toHaveAttribute('href', '/about');
  });
});

describe('snapshots', () => {
  it('Nav snapshot', () => {
    const tree = shallow(<Nav />);
    expect(tree).toMatchSnapshot();
  });
});
