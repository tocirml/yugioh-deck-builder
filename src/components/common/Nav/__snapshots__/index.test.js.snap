// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshots Nav snapshot 1`] = `
<nav
  className="navbar "
  data-testid="navbar"
>
  <ul
    className="navbar-list"
  >
    <li
      className="navbar-list-link"
    >
      <NavLink
        to="/"
      >
        Home
      </NavLink>
    </li>
    <li
      className="navbar-list-link"
    >
      <NavLink
        to="/card-list"
      >
        Card List
      </NavLink>
    </li>
    <li
      className="navbar-list-link"
    >
      <NavLink
        to="/deck-builder"
      >
        Deck Builder
      </NavLink>
    </li>
    <li
      className="navbar-list-link"
    >
      <NavLink
        to="/about"
      >
        About
      </NavLink>
    </li>
  </ul>
</nav>
`;
